fastparse/shared/src/main/scala/fastparse/core/Parsing.scala

```diff
@@ -152,8 +159,12 @@ object Result{
      * the default error message isn't to your liking.
      */
     lazy val trace = {
-      Failure.formatStackTrace(stack, input, index, Failure.formatParser(expected0, index))
+      Failure.formatStackTrace(stack, input, index, Failure.formatParser(expected0, input, index))
     }
+
+    private lazy val lines = input.take(1 + index).lines.toVector
```

Comment:
No need to have it on `TracedFailure`; you already have it on `Failure`, and if you have a `TracedFailure` you have a `Failure`